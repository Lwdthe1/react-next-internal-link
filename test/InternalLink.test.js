import React from "react";
import pjson from "../package.json";
import path from "path";
import renderer from "react-test-renderer";

// Test the root of the lib
const InternalLink = require(path.join(__dirname, "../", pjson.main)).default;

describe.only("InternalLink.test", () => {
  test("Creates internal (relative) link when provided 'as' is an absolute url with protocol and host", () => {
    const component = renderer.create(
      <InternalLink as="http://example.com/collections">
        <a className="link u-paddingLeft2 u-textColorGreenNormal u-border0">
          View collections
        </a>
      </InternalLink>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Creates internal (relative) link when provided 'as' is an absolute url with protocol and host and 'href' is provided", () => {
    const component = renderer.create(
      <InternalLink
        href="/collections?id=abc1"
        as="http://example.com/collection/abc1"
      >
        <a className="link u-paddingLeft2 u-textColorGreenNormal u-border0">
          View collection abc1
        </a>
      </InternalLink>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Creates internal (relative) link with 'href' used as the final anchor href when only 'href' is provided", () => {
    const component = renderer.create(
      <InternalLink href="/collections">
        <a className="link u-paddingLeft2 u-textColorGreenNormal u-border0">
          View all
        </a>
      </InternalLink>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Creates internal (relative) link with 'as' used as the final anchor href when 'href' and 'as' are provided.", () => {
    const component = renderer.create(
      <InternalLink href="/collection?id=abc1" as="collection/abc1">
        <a className="link u-paddingLeft2 u-textColorGreenNormal u-border0">
          View collection abc1
        </a>
      </InternalLink>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Creates internal (relative) link with 'href' used as the final anchor href when 'href' is provided with query params.", () => {
    const component = renderer.create(
      <InternalLink href="/collection?id=abc1">
        <a className="link u-paddingLeft2 u-textColorGreenNormal u-border0">
          View collection abc1
        </a>
      </InternalLink>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Creates internal (relative) link and respects url fragment", () => {
    const component = renderer.create(
      <InternalLink
        href="/collection?id=abc1"
        as="https://example.com/collection/abc1#div=pricing"
      >
        <a
          className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
          style={{ color: "red" }}
          data-field1="field 1"
        >
          View collection abc1
        </a>
      </InternalLink>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Creates internal (relative) link and respects url query string and fragment", () => {
    const component = renderer.create(
      <InternalLink
        href="/collection?id=abc1"
        as="https://example.com/collection/abc1?do=something1&for=me#div=pricing"
      >
        <a
          className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
          style={{ color: "red" }}
          data-field1="field 1"
        >
          View collection abc1
        </a>
      </InternalLink>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Creates internal (relative) link and respects url query string and fragment with port", () => {
    const component = renderer.create(
      <InternalLink
        href="/collection?id=abc1"
        as="http://example.com:3100/collection/abc1?do=something1&for=me#div=pricing"
      >
        <a
          className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
          style={{ color: "red" }}
          data-field1="field 1"
        >
          View collection abc1
        </a>
      </InternalLink>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Creates internal (relative) link and passes along all props", () => {
    const component = renderer.create(
      <InternalLink
        href="/collection?id=abc1"
        as="https://example.com/collection/abc1"
      >
        <a
          className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
          style={{ color: "red" }}
          data-field1="field 1"
        >
          View collection abc1
        </a>
      </InternalLink>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
