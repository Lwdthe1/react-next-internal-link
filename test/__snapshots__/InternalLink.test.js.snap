// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalLink.test Creates internal (relative) link and passes along all props 1`] = `
<a
  className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
  data-field1="field 1"
  href="/collection/abc1"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
>
  View collection abc1
</a>
`;

exports[`InternalLink.test Creates internal (relative) link and respects url fragment 1`] = `
<a
  className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
  data-field1="field 1"
  href="/collection/abc1#div=pricing"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
>
  View collection abc1
</a>
`;

exports[`InternalLink.test Creates internal (relative) link and respects url query string and fragment 1`] = `
<a
  className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
  data-field1="field 1"
  href="/collection/abc1?do=something1&for=me#div=pricing"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
>
  View collection abc1
</a>
`;

exports[`InternalLink.test Creates internal (relative) link and respects url query string and fragment with port 1`] = `
<a
  className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
  data-field1="field 1"
  href="/collection/abc1?do=something1&for=me#div=pricing"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
>
  View collection abc1
</a>
`;

exports[`InternalLink.test Creates internal (relative) link when provided 'as' is an absolute url with protocol and host 1`] = `
<a
  className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
  href="/collections"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  View collections
</a>
`;

exports[`InternalLink.test Creates internal (relative) link when provided 'as' is an absolute url with protocol and host and 'href' is provided 1`] = `
<a
  className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
  href="/collection/abc1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  View collection abc1
</a>
`;

exports[`InternalLink.test Creates internal (relative) link with 'as' used as the final anchor href when 'href' and 'as' are provided. 1`] = `
<a
  className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
  href="/collection/abc1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  View collection abc1
</a>
`;

exports[`InternalLink.test Creates internal (relative) link with 'href' used as the final anchor href when 'href' is provided with query params. 1`] = `
<a
  className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
  href="/collection?id=abc1"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  View collection abc1
</a>
`;

exports[`InternalLink.test Creates internal (relative) link with 'href' used as the final anchor href when only 'href' is provided 1`] = `
<a
  className="link u-paddingLeft2 u-textColorGreenNormal u-border0"
  href="/collections"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  View all
</a>
`;
